File: ios/MendixNative/DevAppMenu/UIAlertActionExt/UIAlertActionExt.m
#import "UIAlertActionExt.h"
#import "objc/runtime.h"

@implementation UIAlertAction (UIAlertActionExt)

- (void) setAccessibilityIdentifier:(NSString *) accessibilityIdentifier {
  objc_setAssociatedObject(self, @"nsh_AccesibilityIdentifier", accessibilityIdentifier, OBJC_ASSOCIATION_RETAIN);
}

- (NSString *) getAccessibilityIdentifier {
  return (NSString *) objc_getAssociatedObject(self, @"nsh_AccesibilityIdentifier");
}

@end

File: ios/Modules/DevAppMenu/UIAlertActionExt/UIAlertActionExt.swift
//import UIKit
//import ObjectiveC
//
//extension UIAlertAction {
//    private static var associatedIdentifierKey: UInt8 = 0 //TODO: verify this
//    
//    @objc public func setAccessibilityIdentifier(_ accessibilityIdentifier: String) {
//        objc_setAssociatedObject(self, &UIAlertAction.associatedIdentifierKey, accessibilityIdentifier, .OBJC_ASSOCIATION_RETAIN)
//    }
//    
//    @objc public func getAccessibilityIdentifier() -> String? {
//        return objc_getAssociatedObject(self, &UIAlertAction.associatedIdentifierKey) as? String
//    }
//}
